operator/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -196,6 +202,43 @@ type ClusterSpec struct {
 	NodePools []NodePoolSpec `json:"nodePools,omitempty"`
 }
 
+// ClusterConfiguration holds values (or references to values) that should be used
+// to configure the cluster. Where the cluster schema defines a non-string type for a
+// given key, the corresponding values here should be string-encoded (according to yaml
+// rules)
+type ClusterConfiguration map[string]ClusterConfigCRDValue
+
+type YAMLRepresentation string
+
+// ClusterConfigCRDValue represents a value of arbitrary type T. Values are string-encoded according to
+// YAML rules in order to preserve numerical fidelity.
+// Because these values must be embedded in a `.bootstrap.yaml` file - during the processing of
+// which, the AdminAPI's schema is unavailable - we endeavour to use yaml-compatible representations
+// throughout. The octet sequence of a Representation will be inserted into a bootstrap template
+// verbatim.
+// TODO: this type should be lifted to a shared module rather than duplicated in the `redpanda` CRD definition.
+type ClusterConfigCRDValue struct {
+	// If the value is directly known, its yaml representation can be embedded here.
+	// Use the string representation of a yaml-serialised value in order to preserve accuracy.
+	// Example:
+	// The string "foo" should be the five octets "\"foo\""
+	// A true value should be the four octets "true".
+	// The number -123456 should be a seven-octet sequence, "-123456".
+	Representation *YAMLRepresentation `json:"repr,omitempty"`
```

Comment:
nit: I think not having matching field names and JSON tags is going to trip someone up along the way. Mind renaming the field to `Repr` (or `Value`) or making everyone writing out YAML have to fully type out `representation`?

Reply:
I'll use Repr rather than Value (I'd like the attribute name to slap people in the face with the expectation here.)